lib/oli/authoring/course.ex

```diff
@@ -63,7 +63,14 @@ defmodule Oli.Authoring.Course do
 
   def get_project!(id), do: Repo.get!(Project, id)
   def get_project_by_slug(nil), do: nil
-  def get_project_by_slug(slug) when is_binary(slug), do: Repo.get_by(Project, slug: slug)
+
+  def get_project_by_slug(slug) when is_binary(slug) do
+    Repo.one(
+      from p in Project,
+      where: p.slug == ^slug and p.status == :active,
```

Comment:
Adding this clause to this existing query fundamentally changes the behavior of `get_project_by_slug/1` to now only work when a project is active.  The following are the places where this method is used:

```
./oli_web/plugs/authorize_project.ex:    case Course.get_project_by_slug(conn.params["project_id"]) do
./oli_web/live/insights/insights.ex:    project = Course.get_project_by_slug(project_slug)
./oli_web/live/curriculum/container/container_live.ex:    project = Course.get_project_by_slug(project_slug)
./oli_web/live/projects/visibility.ex:    project = Course.get_project_by_slug(project_slug)
./oli_web/live/objectives/objectives.ex:    project = Course.get_project_by_slug(Map.get(params, "project_id"))
./oli_web/live/qa/qa_live.ex:    project = Course.get_project_by_slug(project_slug)
./oli_web/live/history/revision_history.ex:    project = Course.get_project_by_slug(project_slug)
./oli_web/live/breadcrumb/breadcrumb_trail_live.ex:    project = Course.get_project_by_slug(project_slug)
./oli_web/controllers/open_and_free_controller.ex:         %{id: project_id} <- Course.get_project_by_slug(project_slug),
./oli_web/controllers/project_controller.ex:    case Course.get_project_by_slug(project_slug) do
./oli_web/controllers/api/resource_controller.ex:    case Course.get_project_by_slug(project_slug) do
./oli_web/controllers/api/resource_controller.ex:    project = Course.get_project_by_slug(project_slug)
./oli_web/controllers/project_plugs.ex:    case Course.get_project_by_slug(conn.params["project_id"]) do
./oli_web/controllers/project_plugs.ex:    case Course.get_project_by_slug(conn.params["project"]) do
./oli/activities.ex:           Course.get_project_by_slug(project_slug)
./oli/activities.ex:           Course.get_project_by_slug(project_slug)
./oli/activities.ex:           Course.get_project_by_slug(project_slug)
./oli/qa/reviewers/accessibility.ex:      Reviews.create_review(Course.get_project_by_slug(project_slug), "accessibility")
./oli/qa/reviewers/pedagogy.ex:    case Reviews.create_review(Course.get_project_by_slug(project_slug), "pedagogy") do
./oli/qa/reviewers/content.ex:    {:ok, review} = Reviews.create_review(Course.get_project_by_slug(project_slug), "content")
./oli/authoring/collaborators.ex:           Course.get_project_by_slug(project_slug)
./oli/authoring/collaborators.ex:           Course.get_project_by_slug(project_slug)
./oli/authoring/collaborators.ex:           Course.get_project_by_slug(project_slug)
./oli/authoring/course.ex:  def get_project_by_slug(nil), do: nil
./oli/authoring/course.ex:  def get_project_by_slug(slug) when is_binary(slug) do
./oli/authoring/editing/page_editor.ex:           {:ok, project} <- Course.get_project_by_slug(project_slug) |> trap_nil(),
./oli/authoring/editing/page_editor.ex:         {:ok, project} <- Course.get_project_by_slug(project_slug) |> trap_nil(),
./oli/authoring/editing/page_editor.ex:         {:ok, project} <- Course.get_project_by_slug(project_slug) |> trap_nil(),
./oli/authoring/editing/activity_editor.ex:    with {:ok, project} <- Course.get_project_by_slug(project_slug) |> trap_nil(),
./oli/authoring/editing/activity_editor.ex:    with {:ok, project} <- Course.get_project_by_slug(project_slug) |> trap_nil(),
./oli/authoring/editing/activity_editor.ex:    with {:ok, project} <- Course.get_project_by_slug(project_slug) |> trap_nil(),
./oli/authoring/editing/activity_editor.ex:           {:ok, project} <- Course.get_project_by_slug(project_slug) |> trap_nil(),
./oli/authoring/editing/activity_editor.ex:           {:ok, project} <- Course.get_project_by_slug(project_slug) |> trap_nil(),
./oli/authoring/editing/activity_editor.ex:      with {:ok, project} <- Course.get_project_by_slug(project_slug) |> trap_nil(),
./oli/authoring/media_library.ex:    case Oli.Authoring.Course.get_project_by_slug(project_slug) do
./oli/authoring/media_library.ex:    project = Oli.Authoring.Course.get_project_by_slug(project_slug)
./oli/authoring/clone.ex:             Course.get_project_by_slug(project_slug) |> Repo.preload(:family) |> trap_nil(),
./oli/qa.ex:    Reviews.delete_reviews(Course.get_project_by_slug(project_slug).id)
./oli/part_components.ex:           Course.get_project_by_slug(project_slug)
./oli/part_components.ex:           Course.get_project_by_slug(project_slug)
./oli/part_components.ex:           Course.get_project_by_slug(project_slug)
```

This change should be reverted, and instead introduce a more explicit way of ensuring that a project is active or deleted, when needed.

Reply:
Reverted call and moved filter into Oli.Plugs.AuthorizeProject